src/main/java/com/example/blogapi/controllers/PostController.java

```diff
@@ -8,6 +20,90 @@
  * de servicio para adaptar el controlador a los endpoints publicados en
  * la API y crear todas las clases que faltan.
  */
+@RestController
+@RequestMapping("/posts")
 public class PostController {
 
+    private PostService postService;
+
+    public PostController(PostService postService) {
+        this.postService = postService;
+    }
+
+    //añade una nueva publicacion
+    @PostMapping("/")
+    public ResponseEntity<?> createPost(@RequestBody CreatePostInput postInput){
+        Post newPost = postService.createPost(postInput);
+        return ResponseEntity.ok(newPost);
+    }
+
+    //buscar publicaciones por filtro
+    @PostMapping("/search")
+    public ResponseEntity<List<Post>> findPost(@RequestBody PostFilter postFilter){
+        List<Post> publicFilter = postService.findPosts(postFilter);
+
+        return ResponseEntity.ok(publicFilter);
+    }
+
+    //Identificador de la publicación donde se desea comentar
+    //duda entre requestparam y pathvariable
+    @PostMapping("/{id}/comments")
+    public ResponseEntity<Comment> addCommentToPost(@RequestParam("id") String postId,
```

Comment:
La notación es incorrecta. Ten en cuenta que `@RequestParam` se usa para extraer atributos del _querystring_, no parámetros de la ruta. Así pues, tu `RequestParam("id")` solo funcionaría si se invocara la URL tal que así: https://localhost:8080/posts/loquesea_nonulo/comments?id=postId. La anotación correcta sería `@PathVariable`

```suggestion
    public ResponseEntity<Comment> addCommentToPost(@PathVariable("id") String postId,
```